### PR TITLE
Ticket6678 razorbill rp100

### PR DESCRIPTION
**Pull request by davidkeymer**
### Description of work

Add Razorbill RP100 PSU

### To test

#6678

### Acceptance criteria

*List the acceptance criteria for the PR*

---

#### Code Review

- [ ] A copy of the manual has been placed on the shared drive
- [ ] Pertitent information has been stored in the [wiki](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Specific-Device-IOC)
- [ ] Does the IOC conform to IBEX standards?
    - [PV naming](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/PV-Naming).
    - [Disable records](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Disable-records)
    - [Record simulation](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Record-Simulation)
    - [Finishing touches](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/IOC-Finishing-Touches)
- [ ] If an OPI has been modified, does it conform to the [style guidelines](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/OPI-Creation)?
- [ ] Do the IOC and support module conform to the new [build guidelines](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Reducing-Build-Dependencies)
- [ ] Have the changes been recorded appropriately in a PR for [release notes](https://github.com/ISISComputingGroup/IBEX/blob/master/release_notes/ReleaseNotes_Upcoming.md)?
- [ ] Is the device's flow control setting correct? [For most devices flow control should be OFF](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Flow-control).

### Functional Tests

- IOC responds correctly in:
    - [ ] Devsim mode
    - [ ] Recsim mode
    - [ ] Real device, if available
- [ ] Supplementary IOCs (`..._0n` where `n>1`) run correctly
- [ ] Log files do not report undefined macros (serach for `macLib: macro` to find instances of `macLib: macro [macro name] is undefined...`

### Final steps

- [ ] Update the IOC submodule in the main EPICS repo. See [Git workflow](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Git-workflow) page for details.
- [ ] Reviewer has merged the associated PR for the [release notes](https://github.com/ISISComputingGroup/IBEX/blob/master/release_notes/ReleaseNotes_Upcoming.md)

### CODE REVIEW

File: RZBRP100/iocBoot/iocRZBRP100-IOC-01/st-common.cmd [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+< $(IOCSTARTUP)/init.cmd
+
+## Device simulation mode IP configuration
+$(IFDEVSIM) drvAsynIPPortConfigure("$(DEVICE)", "localhost:$(EMULATOR_PORT=57677)")
+
+## For recsim:
+$(IFRECSIM) drvAsynSerialPortConfigure("$(DEVICE)", "$(PORT=NUL)", 0, 1, 0, 0)
+
+## For real device:
+$(IFNOTDEVSIM) $(IFNOTRECSIM) drvAsynSerialPortConfigure("$(DEVICE)", "$(PORT=NO_PORT_MACRO)", 0, 0, 0, 0)
```
Comment (Tom-Willemsen) [inline RZBRP100/iocBoot/iocRZBRP100-IOC-01/st-common.cmd:14]: Wiki says this is a USB device but this looks like standard serial port setup - which is it?

File: RZBRP100/iocBoot/iocRZBRP100-IOC-01/config.xml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" ?>
+<ioc_config xmlns:xi="http://www.w3.org/2001/XInclude">
+<config_part>
+<ioc_desc>Razorbill RP100 Strain Cell PSU</ioc_desc>
+<ioc_details></ioc_details>
+<macros>
+<macro name="PORT" pattern="^COM[0-9]+$" description="Serial COM port" />
```
Comment (Tom-Willemsen) [inline RZBRP100/iocBoot/iocRZBRP100-IOC-01/config.xml:7]: Wiki says this is a USB device but this looks like standard serial port setup - which is it?

File: RZBRP100/configure/RELEASE [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+ZLIB=$(SUPPORT)/zlib/master
+
+# optional extra local definitions here
+-include $(TOP)/configure/RELEASE.private
+
+include $(TOP)/../../../ISIS_CONFIG
+-include $(TOP)/../../../ISIS_CONFIG.$(EPICS_HOST_ARCH)
+
+# IOC-specific support module
+RZBRP100=$(SUPPORT)/rzbrp100/main
```
Comment (Tom-Willemsen) [inline RZBRP100/configure/RELEASE:74]: This points at wrong location and causes the IOC to not boot (including during IOC tests)

File: RZBRP100/iocBoot/iocRZBRP100-IOC-01/st-common.cmd (hunk 1 above)
Comment (davidkeymer) [reply RZBRP100/iocBoot/iocRZBRP100-IOC-01/st-common.cmd:14]: Both.  The USB driver creates a virtual COM port on the host, enabling it to be accessed as a standard serial port.  More details have been added to the [Wiki page](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Razorbill-RP100-Strain-Cell-PSU).

File: RZBRP100/iocBoot/iocRZBRP100-IOC-01/config.xml (hunk 2 above)
Comment (davidkeymer) [reply RZBRP100/iocBoot/iocRZBRP100-IOC-01/config.xml:7]: Both.  The USB driver creates a virtual COM port on the host, enabling it to be accessed as a standard serial port.  More details have been added to the [Wiki page](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Razorbill-RP100-Strain-Cell-PSU).

**Review by Tom-Willemsen — APPROVED** [review]

**Pull request merged**